src/js/hooks/productSupplier/form/usePreferenceTypeVariationsFiltering.jsx

```diff
@@ -0,0 +1,32 @@
+import { useState } from 'react';
+
+import _ from 'lodash';
+
+const usePreferenceTypeVariationsFiltering = ({ errors, updatedRows }) => {
+  const [isFiltered, setIsFiltered] = useState(false);
+
+  // react-table triggers filtering on changing value in filter fields (under column header), when
+  // we want to filter after clicking button, we have to trigger changing filters value manually
+  const triggerFiltering = () => [{ value: isFiltered }];
+
+  // If row index is equal to index of invalid row and if the row is dirty
+  // it means it's invalid
+  const isRowValid = (rowIndex) => {
+    const isRowDirty = _.some(Object.values(updatedRows?.[rowIndex] || {}));
```

Comment:
what's the purpose of that method, what does it do behind the scene? An empty object is truthy, so I assume that this method will always return true, hence the `&& isRowDirty` might not make sense?

Reply:
This empty object is a fallback in `Object.values`, so it returns an empty array, and then we are checking if something is true within this array. True means that we entered a value in any field in the row.

Reply:
I didn't realize the fallback is inside the `( )`. I think this shortcut is not intuitive (not passing predicate to the `_.some` function). This is why I hate lodash for methods that are available in the pure JS :smiling_face_with_tear: